plutus-pab/src/Plutus/PAB/Core/ContractInstance/STM.hs

```diff
@@ -149,6 +153,13 @@ awaitSlot targetSlot BlockchainEnv{beCurrentSlot} = do
     guard (current >= targetSlot)
     pure current
 
+-- | Wait until the current time is greater than or equal to the
+-- target time, then return the current time.
+awaitTime :: POSIXTime -> BlockchainEnv -> STM POSIXTime
+awaitTime targetTime be = do
+    let targetSlot = TimeSlot.posixTimeToEnclosingSlot def targetTime
+    TimeSlot.slotToEndPOSIXTime def <$> awaitSlot targetSlot be
```

Comment:
This is ok for now, but in the long run we should just use `threadDelay` or something to wait until the target time, without converting to slots.